### PR TITLE
Generate Beta Packs with New OU Group Logic

### PR DESCRIPTION
**Pull request by JordanBalesBAO**
- added code changes to `packSheets` in order to generate cop25 beta packs
- `getDataPackOrgUnits` now includes a TSNU and HISTORIC PSNU flag to allow for identification of PSNUS that are TSNUS and Countries which were PSNUS and are required as historic PSNUS in Prioritization
- added/modified testing to account for changes
- eliminated tests for cop23 datapack generation - not needed anymore
- updated valid_Org units_2025 rda for COP25 - logic does not affect cop24
-  TESTING: to test run `data-raw/COP25/COP25_Data_Pack_generation_script.R`, should be able to complete all needed 54 countries

### Related Issues
- DP-1375
- etc.

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

File: R/packSheets.R [hunk 1]
```diff
@@ -39,13 +39,49 @@ packDataPackSheets <- function(wb,
...
-      org_units <- getValidOrgUnits(cop_year) %>%
-        dplyr::filter(country_uid %in% country_uids) %>%
-        add_dp_label(orgunits = ., cop_year = cop_year) %>%
-        dplyr::arrange(dp_label) %>%
-        ## Remove DSNUs
-        dplyr::filter(org_type != "DSNU") %>%
-        dplyr::select(PSNU = dp_label, psnu_uid = uid, snu1)
+
+
+      if (cop_year > 2024) {
```
Comment (jason-p-pickering) [inline R/packSheets.R:44]: I am not sure we want > here? This might lead to problems down the road.

File: R/packSheets.R [hunk 2]
```diff
@@ -39,13 +39,49 @@ packDataPackSheets <- function(wb,
...
+
+      # extract historic regions
+      historic_regional_psnus <-
+        getValidOrgUnits(cop_year) %>%
+        dplyr::filter(!is.na(HISTORIC_PSNU)) %>%
```
Comment (jason-p-pickering) [inline R/packSheets.R:49]: Is there a need to use CAPITAL_LETTERS here?

File: R/packSheets.R [hunk 3]
```diff
@@ -111,24 +147,61 @@ packDataPackSheets <- function(wb,
...
+          dplyr::select(PSNU = dp_label, psnu_uid = uid, snu1)
+        if (NROW(temp_org_units_sheet) == 0) {
+          next
+        }
+      } else if (sheet == "AGYW") {
+        temp_org_units_sheet <- getValidOrgUnits(cop_year) %>%
+          dplyr::filter(country_uid %in% country_uids) %>%
+          add_dp_label(orgunits = ., cop_year = cop_year) %>%
+          dplyr::arrange(dp_label) %>%
+          dplyr::filter(!is.na(DREAMS)) %>%
```
Comment (jason-p-pickering) [inline R/packSheets.R:171]: I am not sure we should be using CAPS again here?

File: data-raw/COP25/COP25_Data_Pack_generation_script.R [hunk 4]
```diff
@@ -2,32 +2,70 @@ library(datapackr)
 library(dplyr)
 
 # Point to DATIM login secrets ####
-secrets <- Sys.getenv("SECRETS_FOLDER") %>% paste0(., "datim.json")
+secrets <-
+  Sys.getenv("SECRETS_FOLDER") %>% paste0(., "coptest.json")
```
Comment (jason-p-pickering) [inline data-raw/COP25/COP25_Data_Pack_generation_script.R:6]: Likely this should also be an environment variable

File: tests/testthat/test-getValidOrgUnits.R [hunk 5]
```diff
@@ -10,9 +10,9 @@ test_that("Can error on a missing cop year", {
 
 
 test_that(
-  "Structure of orgunits is the defined",
+  "Structure of orgunits is the defined pre 2025",
   {
-    all_cop_years <- supportedCOPYears()
+    all_cop_years <- supportedCOPYears()[supportedCOPYears() < 2025]
```
Comment (jason-p-pickering) [inline tests/testthat/test-getValidOrgUnits.R:15]: Again, not entirely sure we want to use < as opposed to %in% here in order to be explicit.

File: tests/testthat/test-generateCOP25DataPack.R [hunk 6]
```diff
@@ -142,25 +192,33 @@ with_mock_api({
...
-    #   dplyr::filter(n > 1) %>%
-    #   NROW()
-    #
-    # expect_equal(duplicated_export_rows, 0L)
+    # Malawi requires PNSUS only in prioritization
+    expect_true(all(unlist(
+      purrr::map(sheet_psnus[names(sheet_psnus) == "Prioritization"], identical, wanted_psnus)
+    )))
+
+    # Malwi requires TSNUS which for Malawi are country level for other sheets
```
Comment (jason-p-pickering) [inline tests/testthat/test-generateCOP25DataPack.R:205]: Typo here.

File: R/packSheets.R
Comment (jason-p-pickering) [inline R/packSheets.R:1]: I find the changes here hard to follow, particular the introduction of the concept of "historical" PSNUs

**Review by jason-p-pickering — CHANGES_REQUESTED** [review]
I think this needs a bit more work. Check the upper case variable names and be sure we want to go this route. Typically we have used snake_case. The liberal use of > as opposed to %in% makes me a bit nervous long term.

File: data-raw/COP25/COP25_Data_Pack_generation_script.R (hunk 4 above)
Comment (flopez-bao) [reply data-raw/COP25/COP25_Data_Pack_generation_script.R:6]: Already was there, believe Nemarich added it back in the day for the generation scrips. We use this process in order to swap jsons when using different servers.

File: R/packSheets.R (hunk 1 above)
Comment (flopez-bao) [reply R/packSheets.R]: changed to %in%

File: R/packSheets.R (hunk 3 above)
Comment (flopez-bao) [reply R/packSheets.R:171]: was already capitalized so leaving for now

File: tests/testthat/test-getValidOrgUnits.R (hunk 5 above)
Comment (flopez-bao) [reply tests/testthat/test-getValidOrgUnits.R]: changed to %in%

File: tests/testthat/test-generateCOP25DataPack.R (hunk 6 above)
Comment (flopez-bao) [reply tests/testthat/test-generateCOP25DataPack.R]: fixed

**Comment by flopez-bao** [comment]
- caps were dropped
- operators swapped for `%in%`
- changed `HISTORIC_PSNUS` to `country_was_psnu` flag for a more intuitive name, the flag is similar to way we handle DREAMS for AGYW tab, it marks countries that were historically psnus at the national level like Colombia so we can display it differently in Prioritization

**Review by jason-p-pickering — APPROVED** [review]

**Comment by jason-p-pickering** [comment]
Please fix the conflicts before merging, but that should be straightforward
